charts/pulsar/values.yaml

```diff
@@ -273,12 +273,28 @@ tls:
 auth:
   authentication:
     enabled: false
-    provider: "jwt"
```

Comment:
Since many users don't read release notes, it's useful to add validation to prevent users from upgrading unless they remove deprecated config values.

Add a template called `_values_validation.tpl` and add validation with something like

```
{{- if (and .Values.auth.authentication.enabled (not (empty .Values.auth.authentication.provider))) }}
  {{- fail "ERROR: Setting auth.authentication.provider is no longer supported. For details, see the migration guide in README.md." }}
{{- end }}
```

Reply:
Please also add migration information to README.md